src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+            { symlink_access_key: false, symlink_name: false });
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const account_upgrade_params = { ...new_account_data, identity_path };
+        await create_account_access_keys_index_if_missing(default_config_fs, account_upgrade_params, hidden_old_accounts_path, dbg);
+        await assert_access_key_index_is_updated(new_account_data);
+    });
+});
+
+describe('create_account_name_index_if_missing', () => {
+    beforeAll(async () => {
+        await fail_test_if_default_config_dir_exists('test_config_dir_restructure', default_config_fs);
+    });
+
+    beforeEach(async () => {
+        await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+        await default_config_fs.create_config_dirs_if_missing();
+        await create_fresh_path(default_config_fs.old_accounts_dir_path);
+        await create_fresh_path(hidden_old_accounts_path);
+    }, TEST_TIMEOUT);
+
+    afterEach(async () => {
+        await clean_config_dir(default_config_fs);
+    }, TEST_TIMEOUT);
+
+
+    it('create_account_name_index_if_missing() - single regular account', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: false });
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined,
+            { symlink_access_key: false, symlink_name: false });
+
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const account_upgrade_params = { ...new_account_data, account_name: new_account_data.name, identity_path };
+        await create_account_name_index_if_missing(default_config_fs, account_upgrade_params, dbg);
+        await assert_name_index_is_updated(new_account_data);
+    });
+
+    it('create_account_name_index_if_missing() - anonymous account', async () => {
+        const new_account_data = { _id: String(1), name: 'anonymous', user: 'root' };
+        const symlink_options = { symlink_access_key: false };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, symlink_options);
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined, symlink_options);
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const account_upgrade_params = { ...new_account_data, account_name: new_account_data.name, identity_path };
+        await create_account_name_index_if_missing(default_config_fs, account_upgrade_params, dbg);
+        await assert_name_index_is_updated(new_account_data);
+    });
+
+    it('create_account_name_index_if_missing() - new name index already exists', async () => {
```

Comment:
What's different in this case? ("new name index already exists" what should we expect to see?)

Reply:
from "new name index already exists and points to a non existing location"?
this is a re-entrancy test, it simulates a situation where we already have the /etc/noobaa.conf.d/accounts_by_name/account_name.symlink but it points to an incorrect target, after the call to create_account_name_index_if_missing() we expect it will point to the correct location